EFM/source/RAPTOR.cpp

```diff
@@ -0,0 +1,1541 @@
+bool ed_fm_add_local_moment_component(double& x, double& y, double& z) { return false; }
+bool ed_fm_add_global_moment_component(double& x, double& y, double& z) { return false; }
+bool ed_fm_enable_debug_info() { return false; }
+
+//########################################################################################################
+//########################################################################################################
+//########################################################################################################
+
+    // ----- Cockpit Logic -----
+
+inline float limit(float value, float min_val, float max_val) {
+    if (value < min_val) return min_val;
+    if (value > max_val) return max_val;
+    return value;
+}
+
+CockpitManager::CockpitManager() {
+    draw_args_[709] = 0.0f;
+    draw_args_[712] = 0.0f;
+    draw_args_[713] = 0.0f;
+    draw_args_[714] = 0.0f;
+    draw_args_[715] = 0.0f;
+}
+
+void CockpitManager::initialize() {
+    // Initialize parameter handles
+    param_handles_["APU_POWER"] = param_api_.getParamHandle("APU_POWER");
+    param_handles_["BATTERY_POWER"] = param_api_.getParamHandle("BATTERY_POWER");
+    param_handles_["MAIN_POWER"] = param_api_.getParamHandle("MAIN_POWER");
+    param_handles_["WoW"] = param_api_.getParamHandle("WoW");
+    param_handles_["AAR"] = param_api_.getParamHandle("AAR");
+    param_handles_["N_GEAR_LIGHT"] = param_api_.getParamHandle("N_GEAR_LIGHT");
+    param_handles_["R_GEAR_LIGHT"] = param_api_.getParamHandle("R_GEAR_LIGHT");
+    param_handles_["L_GEAR_LIGHT"] = param_api_.getParamHandle("L_GEAR_LIGHT");
+    param_handles_["TAXI_SWITCH"] = param_api_.getParamHandle("TAXI_SWITCH");
+    param_handles_["NAV_LIGHT_SWITCH"] = param_api_.getParamHandle("NAV_LIGHT_SWITCH");
+    param_handles_["FORM_KNOB"] = param_api_.getParamHandle("FORM_KNOB");
+    param_handles_["AAR_KNOB"] = param_api_.getParamHandle("AAR_KNOB");
+    param_handles_["AAR_READY"] = param_api_.getParamHandle("AAR_READY");
+
+    // Initialize draw arguments based on Lua parameters
+    draw_args_[709] = static_cast<float>(getParameter("TAXI_SWITCH", 0.0)); // TAXI_SWITCH
+    draw_args_[712] = static_cast<float>(getParameter("AAR", 0.0)); // AAR port
+    draw_args_[713] = static_cast<float>(getParameter("AAR_KNOB", 0.0)); // AAR_KNOB
+    draw_args_[714] = static_cast<float>(getParameter("FORM_KNOB", 0.0)); // FORM_KNOB
+    draw_args_[715] = static_cast<float>(getParameter("NAV_LIGHT_SWITCH", 0.0)); // NAV_LIGHT_SWITCH
+
+    // Set default power states
+    param_api_.setParamNumber(param_handles_["MAIN_POWER"], (RAPTOR::left_engine_state == RAPTOR::RUNNING || RAPTOR::right_engine_state == RAPTOR::RUNNING) ? 1.0 : 0.0);
+    param_api_.setParamNumber(param_handles_["BATTERY_POWER"], (RAPTOR::left_engine_switch || RAPTOR::right_engine_switch) ? 1.0 : 0.0);
+}
+
+void CockpitManager::update(double dt) {
```

Comment:
Could be an idea to split this into another file as this file is already decently big.